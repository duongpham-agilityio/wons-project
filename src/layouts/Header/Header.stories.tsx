import { Meta, StoryObj } from '@storybook/react';
import { NextUIProvider } from '@nextui-org/react';
import { ThemeProvider as NextThemesProvider } from 'next-themes';
import Header from '.';

export default {
  title: 'Layouts/Header',
  component: Header,
  decorators: [
    (Story) => (
      <NextUIProvider>
        <NextThemesProvider attribute="class" defaultTheme="light">
          <Story />
        </NextThemesProvider>
      </NextUIProvider>
    ),
  ],
} as Meta;

type Story = StoryObj<typeof Header>;

// Header type default
export const HeaderDefault: Story = {
  args: {},
};
