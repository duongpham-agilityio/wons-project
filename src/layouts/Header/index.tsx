import { ButtonCustom, InputCustom } from '@/components';
import { Logout, NotificationIcon, SearchIcon } from '@/icons';
import { handleLogout } from '@/utils';

const Header = () => {
  return (
    <header className="flex justify-between w-full h-[104px] bg-white border-b-2 border-gray-50 pl-6 pr-4xl dark:bg-dark-900 dark:border-gray-700">
      <div className="flex items-center relative w-xl">
        <InputCustom
          color="default"
          placeholder="search"
          size="xs"
          radius="xs"
          startContent={
            <SearchIcon
              width="21px"
              height="21px"
              className="mr-5 cursor-pointer"
            />
          }
        />
      </div>
      <div className="flex items-center">
        <NotificationIcon
          width="21px"
          height="26px"
          className="mr-10 cursor-pointer"
        />
        <div className="flex items-center border-l-2 border-gray-50 pl-10 dark:border-gray-700">
          <span className="rounded-full w-sm h-sm bg-translucent-400 text-blue-400 flex items-center justify-center text-xl font-semibold">
            TE
          </span>
          <div className="flex flex-col ml-6 font-semibold">
            <span className="text-xl text-dark-800 dark:text-white">
              Theo Edwards
            </span>
            <span className="text-base text-grayBlue-400">Administrator</span>
          </div>
          <ButtonCustom
            color="unstyled"
            size="unstyled"
            startContent={<Logout width="28px" height="28px" />}
            aria-label="Logout Button"
            onClick={handleLogout}
          />
        </div>
      </div>
    </header>
  );
};

export default Header;
