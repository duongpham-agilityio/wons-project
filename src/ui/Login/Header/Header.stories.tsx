import { Meta, StoryObj } from '@storybook/react';
import Header from '.';

export default {
  title: 'UI/Login/HeaderLogin',
  component: Header,
} as Meta;

type Story = StoryObj<typeof Header>;

// Header login
export const HeaderLoginDefault: Story = {
  args: {},
};
