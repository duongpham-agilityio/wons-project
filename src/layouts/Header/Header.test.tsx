import { render, screen } from '@testing-library/react';
import Header from '.';

describe('Header Layout', () => {
  test('render header', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test('renders search input', () => {
    render(<Header />);

    const searchPlaceholder = screen.getByPlaceholderText('search');
    expect(searchPlaceholder).toBeInTheDocument();
  });

  test('renders user information', () => {
    render(<Header />);

    const userName = screen.getByText('Theo Edwards');
    expect(userName).toBeInTheDocument();
    const userRole = screen.getByText('Administrator');
    expect(userRole).toBeInTheDocument();
  });
});
