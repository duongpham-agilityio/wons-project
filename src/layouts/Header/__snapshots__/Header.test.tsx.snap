// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Layout render header 1`] = `
<div>
  <header
    class="flex justify-between w-full h-[104px] bg-white border-b-2 border-gray-50 pl-6 pr-4xl dark:bg-dark-900 dark:border-gray-700"
  >
    <div
      class="flex items-center relative w-xl"
    >
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end"
        data-filled="true"
        data-filled-within="true"
        data-slot="base"
      >
        <div
          class="flex flex-col w-xl h-14"
          data-slot="main-wrapper"
        >
          <div
            class="relative tap-highlight-transparent flex-row shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 flex items-center space-x-2 h-full dark:bg-gray-700 focus-within:ring-0 py-4 px-6 w-full rounded-lg"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <div
              class="w-full items-center h-full box-border flex"
              data-slot="inner-wrapper"
            >
              <svg
                class="mr-5 cursor-pointer"
                fill="none"
                height="21px"
                viewBox="0 0 22 22"
                width="21px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.25 20.9878C4.6 20.9878 0 16.3878 0 10.7378C0 5.08779 4.6 0.487793 10.25 0.487793C15.9 0.487793 20.5 5.08779 20.5 10.7378C20.5 16.3878 15.9 20.9878 10.25 20.9878ZM10.25 1.98779C5.42 1.98779 1.5 5.91779 1.5 10.7378C1.5 15.5578 5.42 19.4878 10.25 19.4878C15.08 19.4878 19 15.5578 19 10.7378C19 5.91779 15.08 1.98779 10.25 1.98779Z"
                  fill="#809FB8"
                  fill-rule="evenodd"
                />
                <path
                  d="M20.5502 21.9877C20.3602 21.9877 20.1702 21.9177 20.0202 21.7677L17.0202 18.7677C16.7302 18.4777 16.7302 17.9977 17.0202 17.7077C17.3102 17.4177 17.7902 17.4177 18.0802 17.7077L21.0802 20.7077C21.3702 20.9977 21.3702 21.4777 21.0802 21.7677C20.9302 21.9177 20.7402 21.9877 20.5502 21.9877Z"
                  fill="#809FB8"
                />
              </svg>
              <input
                aria-label="search"
                class="w-full bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-grayBlue-400 flex-grow focus:outline-none focus:ring-0 text-base font-semibold"
                data-has-start-content="true"
                data-slot="input"
                id="react-aria-:r0:"
                placeholder="search"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center"
    >
      <svg
        class="mr-10 cursor-pointer"
        fill="none"
        height="26px"
        viewBox="0 0 21 27"
        width="21px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.35046 25.1675C6.85608 24.6702 6.48401 24.065 6.26343 23.3995C4.91614 23.1778 3.58887 22.8484 2.29443 22.4145C1.32617 22.1082 0.549683 21.3781 0.18457 20.4305C-0.158203 19.4783 -0.0384521 18.421 0.508301 17.5695L1.88647 15.2695C2.21289 14.6481 2.40125 13.9635 2.43848 13.2625V9.79255C2.4408 6.42719 4.51538 3.41195 7.65649 2.20722C8.23096 1.14647 9.34131 0.486679 10.5477 0.489265C11.754 0.491852 12.8616 1.1564 13.4315 2.2196C16.5554 3.4336 18.6141 6.44012 18.6166 9.79156V13.2655C18.657 13.9701 18.845 14.6583 19.1686 15.2855L20.5345 17.5685C21.0492 18.4336 21.1462 19.4848 20.7985 20.4295C20.4553 21.3697 19.7003 22.1008 18.7496 22.4135C17.46 22.8473 16.1373 23.1763 14.7947 23.3975C14.3096 24.8814 13.0876 26.0062 11.5688 26.3672C10.0499 26.7282 8.45264 26.2734 7.35168 25.1665L7.35046 25.1675ZM8.62146 23.8935C9.16333 24.439 9.91309 24.7261 10.6807 24.6822C11.4484 24.6382 12.1604 24.2673 12.6365 23.6635C11.9352 23.7201 11.2322 23.7485 10.5275 23.7485C9.82275 23.7485 9.11975 23.7201 8.41846 23.6635C8.48169 23.7436 8.54956 23.8201 8.62146 23.8925V23.8935ZM4.23547 9.79366V13.2676C4.19336 14.2931 3.9198 15.2958 3.43542 16.2006L2.05737 18.5006C1.79797 18.8856 1.73157 19.3689 1.8772 19.8096C2.05347 20.2441 2.42078 20.5727 2.87231 20.6996C7.8429 22.3822 13.2289 22.3822 18.1995 20.6996C18.6292 20.5578 18.9694 20.2257 19.1213 19.7996C19.2817 19.3707 19.2376 18.8919 19.0012 18.4996L17.6233 16.1996C17.136 15.2913 16.8623 14.2837 16.8232 13.2536V9.79266C16.8273 6.65929 14.5297 3.9984 11.4291 3.5457C10.811 3.46886 10.1855 3.47223 9.56824 3.55569C6.4978 4.03872 4.2356 6.68449 4.23547 9.79266V9.79366Z"
          fill="#809FB8"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="flex items-center border-l-2 border-gray-50 pl-10 dark:border-gray-700"
      >
        <span
          class="rounded-full w-sm h-sm bg-translucent-400 text-blue-400 flex items-center justify-center text-xl font-semibold"
        >
          TE
        </span>
        <div
          class="flex flex-col ml-6 font-semibold"
        >
          <span
            class="text-xl text-dark-800 dark:text-white"
          >
            Theo Edwards
          </span>
          <span
            class="text-base text-grayBlue-400"
          >
            Administrator
          </span>
        </div>
        <svg
          class="ml-l cursor-pointer"
          fill="none"
          height="21px"
          viewBox="0 0 12 23"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.29044 20.8966L10.4455 16.7626C10.7523 16.4572 11.2484 16.4572 11.5553 16.7626C11.8589 17.0663 11.8589 17.5587 11.5553 17.8626L6.84549 22.5486C6.53824 22.8534 6.04263 22.8534 5.7355 22.5486L1.02554 17.8626C0.721954 17.5588 0.721954 17.0664 1.02554 16.7626C1.33243 16.4572 1.82852 16.4572 2.13541 16.7626L6.29044 20.8966ZM6.29044 2.67658L10.4455 6.81158C10.7526 7.11638 11.2482 7.11638 11.5553 6.81158C11.8584 6.50758 11.8586 6.01564 11.5555 5.71158L6.84549 1.02458C6.53799 0.720323 6.04288 0.720323 5.7355 1.02458L1.02554 5.71158C0.722443 6.01561 0.722443 6.50755 1.02554 6.81158C1.33267 7.11638 1.82828 7.11638 2.13541 6.81158L6.29044 2.67658Z"
            fill="#D9E1E7"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </header>
</div>
`;
